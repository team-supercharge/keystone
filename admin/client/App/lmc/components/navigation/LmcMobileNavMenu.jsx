import React, { Component } from 'react'
import PropTypes from 'prop-types'
import LmcMobileNavSection from './LmcMobileNavSection.jsx'

export default class LmcMobileNavMenu extends Component {
    isActive = (url) => {
        const pathname = this.props.location.pathname
        if (!url) {
            return pathname === `${Keystone.adminPath}/` ? true : false
        }
        if (pathname && url) {
            return pathname.match(url) ? true : false
        }
    }

    render () {
        const { sections, toggleMenu, location } = this.props
        if (!sections || !sections.length) return null

        return (
            <nav className="MobileNavigation__menu">
                <div className="MobileNavigation__sections">
                    { sections.map((section, i) => {
                        const url = `${Keystone.adminPath}/${section.path}`
                        const className = this.isActive(section.path) 
                            ? 'MobileNavigation__section is-active' 
                            : 'MobileNavigation__section'

                        return (
                            <LmcMobileNavSection
                                key={i}
                                className={className}
                                url={url}
                                items={section.items}
                                label={section.label}
                                location={location}
                                onClick={toggleMenu}
                            />
                        )
                    }) }
                </div>
            </nav>
        )
    }
}

LmcMobileNavMenu.propTypes = {
    location: PropTypes.object.isRequired,
    sections: PropTypes.array.isRequired
}