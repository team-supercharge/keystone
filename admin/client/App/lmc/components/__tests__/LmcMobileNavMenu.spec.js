import React from 'react'
import { shallow } from 'enzyme'
import LmcMobileNavMenu from '../navigation/LmcMobileNavMenu.jsx'

describe('LmcMobileNavMenu', () => {
    let wrapper 
    let sections
    let location

    beforeEach(() => {
        sections = []
        location = {}
        
        wrapper = shallow(
            <LmcMobileNavMenu
                sections={sections}
                location={location}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
})