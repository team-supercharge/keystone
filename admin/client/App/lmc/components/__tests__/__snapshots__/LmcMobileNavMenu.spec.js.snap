// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcMobileNavMenu renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": null,
  "nodes": Array [
    null,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcMobileNavMenu
        location={Object {}}
        sections={Array []}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcMobileNavMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "isActive": [Function],
        "props": Object {
          "location": Object {},
          "sections": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": null,
        "_debugID": 2,
        "_renderedOutput": null,
      },
      "_renderedNodeType": 2,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcMobileNavMenu
    location={Object {}}
    sections={Array []}
  />,
}
`;
